app/controllers/users_controller.rb

```diff
@@ -4,10 +4,10 @@ def create
     @user = User.new(user_params)
 
     if @user.save
-      redirect_to root_path, notice: 'User successfully created.'
       session[:user_id] = @user.id
+      redirect_to root_path, notice: 'User successfully created.'
     else
-      render :new
+      render :new, notice: 'Error: Unable to create user account'
```

Comment:
Not sure how to cause this error case, as it is the result of a failed save. I'll change it to the flash error for now.